lib/client/cluster_client.go

```diff
@@ -121,23 +83,32 @@ func (c *ClusterClient) SessionSSHConfig(ctx context.Context, user string, targe
 	}
 
 	mfaClt := c
-	if params.RouteToCluster != rootClusterName {
-		jumpHosts := c.tc.JumpHosts
-		// In case of MFA connect to root teleport proxy instead of JumpHost to request
-		// MFA certificates.
-		c.tc.JumpHosts = nil
-		clt, err := c.tc.ConnectToCluster(ctx)
-		c.tc.JumpHosts = jumpHosts
-		if err != nil {
+	closer := func() error { return nil }
+	if target.Cluster != rootClusterName {
+		if err := c.tc.WithoutJumpHosts(func(tcNoJump *TeleportClient) error {
+			clt, err := c.tc.ConnectToCluster(ctx)
```

Comment:
```suggestion
			clusterClient, err := c.tc.ConnectToCluster(ctx)
```

(or whatever specific this is, it it's not that)